### PR TITLE
Feature: Withdraw Confirmation mobile modal

### PR DESCRIPTION
**Pull request by josadcha**
https://jira.openware.work/browse/CORE-369
Screenshot 2020-09-25 at 14 34 27

### CODE REVIEW

File: src/mobile/components/ModalWithdrawConfirmation/index.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            </div>
+        );
+    };
+
+    return (
+        <Modal title={renderHeader()} onClose={props.onDismiss} isOpen={props.show}>
+            <div>
+                {renderBody()}
+            </div>
+            <div>
```
Comment (akhlopiachyi) [inline src/mobile/components/ModalWithdrawConfirmation/index.tsx:91]: i guess you can remove this div

**Review by akhlopiachyi — APPROVED** [review]

File: src/mobile/components/ModalWithdrawConfirmation/index.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    onDismiss: () => void;
+    rid: string;
+    show: boolean;
+    precision: number;
+}
+
+const ModalWithdraw = (props: ModalWithdrawConfirmationProps) => {
+    const intl = useIntl();
+
+    const translate = (e: string) => {
```
Comment (VladyslavP) [inline src/mobile/components/ModalWithdrawConfirmation/index.tsx:20]: why do we need this function?
let's use `formatMessage`

`const { formatMessage } = useIntl();`

and use it everywhere instead

File: src/mobile/components/ModalWithdrawConfirmation/index.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        return intl.formatMessage({id: e});
+    };
+
+    const renderHeader = () => {
```
Comment (VladyslavP) [inline src/mobile/components/ModalWithdrawConfirmation/index.tsx:24]:
```
const renderHeader = React.useCallback(() =>
 <div className="pg-exchange-modal-submit-header">
        {formatMessage({ id: 'page.mobile.wallet.withdraw.modal.confirmation' })}
 </div>
}, [formatMessage]);
```


We agreed to use useCallback for every function in functional components

File: src/mobile/components/ModalWithdrawConfirmation/index.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        return (
+            <div className="pg-exchange-modal-submit-header">
+                {translate('page.mobile.wallet.withdraw.modal.confirmation')}
+            </div>
+        );
+    };
+
+    const renderBody = () => {
+        const { amount, currency, precision, rid } = props;
+        const formattedCurrency = currency.toUpperCase();
```
Comment (VladyslavP) [inline src/mobile/components/ModalWithdrawConfirmation/index.tsx:34]: we use this variable just once, no need for creating it, use **currency.toUpperCase()** in the code

File: src/mobile/components/ModalWithdrawConfirmation/index.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+    const renderHeader = () => {
+        return (
+            <div className="pg-exchange-modal-submit-header">
+                {translate('page.mobile.wallet.withdraw.modal.confirmation')}
+            </div>
+        );
+    };
+
+    const renderBody = () => {
```
Comment (VladyslavP) [inline src/mobile/components/ModalWithdrawConfirmation/index.tsx:32]: wrap it with useCallback

File: src/mobile/components/ModalWithdrawConfirmation/index.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+                    <span className="mobile-modal-body__withdraw-confirm--mute">
+                        {translate('page.mobile.wallet.withdraw.modal.confirmation.message2')}
+                    </span>
+                    <span className="mobile-modal-body__withdraw-confirm--light">{rid}</span>
+                </div>
+            </div>
+        );
+    };
+
+    const renderFooter = () => {
```
Comment (VladyslavP) [inline src/mobile/components/ModalWithdrawConfirmation/index.tsx:61]: useCallback

**Review by VladyslavP — CHANGES_REQUESTED** [review]

File: src/mobile/components/ModalWithdrawConfirmation/index.tsx [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+                    {formatMessage({ id: 'page.body.wallets.tabs.withdraw.modal.button.withdraw' })}
+                </Button>
+            </div>
+        );
+    }, [formatMessage, onSubmit, onDismiss]);
+
+    return (
+        <Modal title={renderHeader()} onClose={onDismiss} isOpen={props.show}>
+            {renderBody()}
+            {renderFooter()}
```
Comment (berezh) [inline src/mobile/components/ModalWithdrawConfirmation/index.tsx:90]: Fine, but usually, functions are used when there is some logic, for example show or not show. In this case there is no any logic, so you can put that tags inside <Modal/> wihout wrapping them in methods. The code will be more clear, IHMO.

**Review by berezh — APPROVED** [review]

**Review by VladyslavP — APPROVED** [review]

**Pull request merged**